### ISSUE TITLE
Additional range types for versions conforming to the RPM and Debian versioning standards

### ISSUE THREAD

**Issue by andrewpollock** [ossf/osv-schema#252]
Would it make sense to add additional range types for RPM and dpkg/Debian versioning standards?

The benefit being that the expectations and treatment of these ranges would be explicit.

References:
- https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/
- https://rpm-software-management.github.io/rpm/manual/spec.html
- https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-version

**Comment by jasinner** [comment]
Makes sense to me. A reference implementation of comparing RPM versions can be found in the [RPM source code](https://github.com/rpm-software-management/rpm/blob/master/rpmio/rpmvercmp.c). 

For clients using Go there is an implementation here: https://github.com/knqyf263/go-rpm-version

**Comment by oliverchang** [comment]
Hmm, the intention was that the `ECOSYSTEM` range would cover these. There are a lot of different versioning schemes out there, and we made the explicit decision to avoid having to catalogue all of these, particularly when they're tied to the underlying ecosystem already.
